### DISCUSSION TITLE
March Release - Introducing our new Map component

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#5286] [Announcements]
Since the launch of our Cloud platform in November, over 20,000 companies from over 100 countries are now using Budibase. This does not include open-source users - which, we believe, are greater in number. These figures are astonishing, and it's a beautiful feeling to stop and consider how Budibase is sparking joy in so many users’ lives. This is why we dedicated this release to quality-of-life changes and a new map component!

### World Map Component
You can now build digital maps within Budibase. Functionality includes:
- Displaying latitude and longitude coordinates
- Displaying markers on maps
- Assigning actions to markers
- Customizing your map with different tilesets
- Allowing users to zoom in and out
- Marking the user’s current location on the map

Find the new Map component documentation here:
[https://docs.budibase.com/docs/map](https://docs.budibase.com/docs/map)

![map](https://user-images.githubusercontent.com/49767913/162157077-0b658225-f4eb-4e7c-b860-c97fff9c7211.gif)

---

### “Continue If” Action
We’ve introduced conditional steps to the [Actions](https://docs.budibase.com/docs/actions) UI - this allows you to halt the execution of a chain of actions if a specific condition is met.
![conintue if](https://user-images.githubusercontent.com/49767913/162157339-3c5367a8-f2e8-4d7f-8b66-1daadbfcdff2.png)

---

### Templates UI refresh
Templates now have a new home within the builder, and users can now select from over 50 free templates.
![templates](https://user-images.githubusercontent.com/49767913/162161490-3f02058a-fb13-421f-acba-4bbf7b6a8528.png)

---

### Improved component tree scrolling
Budibase users reported having trouble using the platform when their component tree grew to a specific size. We fixed this with a brand new, smooth scrolling experience that will make working with the tree a much more pleasant experience. When selecting a component in the preview, the tree will scroll and present the selected component.
![ezgif com-gif-maker (10)](https://user-images.githubusercontent.com/49767913/162161676-8b478bb6-c038-4fbd-bb4b-add23a7ab82e.gif)

---

### Additional improvements
- Added the ability to duplicate screens
- Copying and pasting or duplicating components (and now screens) will keep all data bindings functional (including JS)
- Added the ability to delete all screens underneath a route folder
- Added the selected component's type to the settings panel
- Adjust table height calculation slightly to avoid scrolling when not needed
- Refactored some of the new screen modal logic to make it cleaner, fix some rare bugs and be reusable (for example for duplicating screens)

---

### Bug Fixes
- Charts not rerendering after settings updates
- Date filtering
- Pagination on REST responses 
- Google sheets authentication bug
- Support for dark theme in apex charts
- SQL parsing fixes
- NGINX CSP Policy updates
- Fixed some issues with public apps

---

### Coming up in the April release
- Google sheets integration
- Foreach automation block
- Firebase integration
- Secrets management
- UX updates

**Comment by gerhardmpl** [comment]
A lot of great new features and improvements coming with the March release - nice 👍

**Reply by ryzam** [reply]
[Request] - Navigation link with icon, looks more professional layout page

**Comment by samg666** [comment]
Thanks for the great work. The Copying and pasting enhancement help a lot, we can much easier to re-use the existing componments now.

**Comment by poirazis** [comment]
Thank you for this great product and congrats on your growth. I wish you the very best!

**Comment by tobehn** [comment]
This is huge again - thank you so much !!!

I'm very taken with how you guys are implementing the community's suggestions.

**Comment by theobouwman** [comment]
Im on version 1.0.110 but don't see the map component...

**Reply by tobehn** [reply]
I guess you have to update your app. 

1. Go into you App where you want to insert the map component
2. press the update icon upper right corner
![grafik](https://user-images.githubusercontent.com/40893107/162989511-0b304212-a4dd-42ad-be51-d85b31801523.png)
3. Pop Up - Press update
![grafik](https://user-images.githubusercontent.com/40893107/162989801-3038a7fe-cc5d-4af1-91f5-94adf07fb06a.png)


After that you should be able to insert the map component

**Reply by theobouwman** [reply]
![Screenshot 2022-04-12 at 17 41 48](https://user-images.githubusercontent.com/16098190/163001123-6e381d50-148b-4911-bd6c-8a790cb857c9.png)

**Reply by tobehn** [reply]
And you still cannot see the element at the bottom of the List? in the Design panel of your App?

![grafik](https://user-images.githubusercontent.com/40893107/163003429-ba21efd7-b6f5-488e-ac91-72fe5cadab55.png)

Did you follow this guide?
https://docs.budibase.com/docs/map

**Reply by theobouwman** [reply]
Mh weird. I think it was something with cache. It works now!
